app/js/lib/foxbox/services.js

```diff
@@ -0,0 +1,246 @@
+import EventDispatcher from './event-dispatcher';
+import BaseService from './services/base';
+import IpCameraService from './services/ip-camera';
+import LightService from './services/light';
+import DoorLockService from './services/door-lock';
+import MotionSensorService from './services/motion-sensor';
+
+const p = Object.freeze({
+  api: Symbol('api'),
+  settings: Symbol('settings'),
+  db: Symbol('db'),
+  cache: Symbol('services'),
+  isPollingEnabled: Symbol('isPollingEnabled'),
+  nextPollTimeout: Symbol('nextPollTimeout'),
+
+  // Private methods.
+  schedulePoll: Symbol('schedulePoll'),
+  getServiceInstance: Symbol('getServiceInstance'),
+  hasDoorLockChannel: Symbol('hasDoorLockChannel'),
+  updateServiceList: Symbol('updateServiceList'),
+  getCache: Symbol('getCache'),
+});
+
+/**
+ * Compare 2 objects. Returns true if all properties of object A have the same
+ * value in object B. Extraneous properties in object B are ignored.
+ * Properties order is not important.
+ *
+ * @param {Object} objectA
+ * @param {Object} objectB
+ * @return {boolean}
+ */
+const isSimilar = (objectA, objectB) => {
+  const keysA = Object.keys(objectA);
+  const keysB = Object.keys(objectB);
+
+  if (keysA.length !== keysB.length) {
+    return false;
+  }
+
+  return !keysA.some((keyA) => {
+    const valueA = objectA[keyA];
+    const valueB = objectB[keyA];
+
+    const typeA = typeof valueA;
+
+    if (typeA !== typeof valueB) {
+      return true;
+    }
+
+    if (typeA !== 'object' || valueA === null || valueB === null) {
+      return valueA !== valueB;
+    }
+
+    return !isSimilar(valueA, valueB);
+  });
+};
+
+export default class Services extends EventDispatcher {
```

Comment:
I'm not a big fan of changing the signature of the `foxbox` instance and use `foxbox.service` for everything related to the services, Splitting the logic in multiple files makes sense for the code organisation but that shouldn't impact the publicly exposed API. It seems longer and unnecessarily complicated to me. Or maybe I just need to get used to it.
What do you think?
What else is exposed from the `foxbox` instance? (e.g. `foxbox.boxes`, `foxbox.clear()`...)?